File: CRMSystem/CrmSystem.DAL/Entities/ShortLink.cs
using System.ComponentModel.DataAnnotations;
using System.ComponentModel.DataAnnotations.Schema;

namespace CrmSystem.DAL.Entities
{
    /// <summary>
    /// Сокращенная ссылка
    /// </summary>
    public class ShortLink
    {
        // Идентификатор
        [Key]
        [Required]
        [DatabaseGenerated(DatabaseGeneratedOption.Identity)]
        public int ShortLinkId { get; set; }

        // Короткая ссылка
        [Required]
        public string Short { get; set; }

        // Полная ссылка
        [Required]
        public string Full { get; set; }

        // Владелец ссылки
        public string OwnerId { get; set; }

        // Компания к которой относится вокращенная ссылка
        public int? CompanyId { get; set; }
    }
}

File: crmsystem-client/src/app/models/shortLink.ts
import { Employee } from './employeeModel';
import { CompanyModel } from './companyModel';

export class ShortLink {
    shortLinkId: number;
    short: string;
    full: string;
    owner: Employee;
    ownerId: string;
    company: CompanyModel;
    companyId: number;
}